packages/fullscreenView/components/FullscreenView.tsx

```diff
@@ -16,32 +14,34 @@ export interface FullscreenViewProps {
   title: React.ReactNode;
   /** The subtitle that appears in the header */
   subtitle?: React.ReactNode;
-  /** Whether we automatically add padding to the body of the modal. */
+  /** Function that gets called when the modal is closed */
+  onClose: (event?: React.SyntheticEvent<HTMLElement>) => void;
+}
+export interface FullscreenViewProps extends HeaderProps {
   isContentFlush?: boolean;
   /** Custom header content component. ⚠️Use rarely and with caution⚠️ */
-  headerComponent?: React.ReactNode;
+  headerComponent?: React.JSXElementConstructor<HeaderProps>;
```

Comment:
I'm not really familiar with `React.JSXElementConstructor`, but could this be a breaking change? Say if someone was using a component for the header that didn't have `HeaderProps`?

Reply:
So, from my understanding `React.ReactNode` was the incorrect type altogether for what users were meant to be passing in (and what it did with `headerComponent`). What we want is a React component declaration: some function that will return JSX. That component will only ever get HeaderProps, but it's possible that someone would be using it with a component that took other stuff. So, I could see where this might be too strict.

Our other option is `React.JSXElementConstructor<any>`. What else could I possibly do?

Reply:
I'm actually getting a TS error trying to use the `headerComponent` prop in kommander-ui (with and w/o your changes). Maybe we could take a look in our 1:1.

Reply:
We looked and resolved the TS error in kommander-ui. Discussed and opted for an any here instead just to make sure we don't introduce a breaking change.

Reply:
❓ Why don't we want breaking changes? If that change would be meaningful we should do it? (With the right commit message to trigger a breaking change release of course)

Reply:
breaking changes are fine but I was hoping to batch them in a forthcoming release. Seems like overkill in this case to introduce it when we are not even sure if we want to use this feature of this component in the future. I think the better breaking change would be to just remove this functionality since Kommander UI doesn't leverage it at all